text/0000-disjoint-groups/0000-disjoint-groups.md

```diff
@@ -0,0 +1,149 @@
+
+### Data chains
+
+The current group definition would considerably complicate [data chains](https://github.com/dirvine/data_chain): With disjoint groups, all members of a group can sign the same link block - consising of all the group members -, and are responsible for exactly the same set of data chunks, so they can sign the same data blocks.
+
+
+## Detailed design
+
+The definition of a close group is modified so that at each point in time, the name space is partitioned into disjoint groups. Each group is defined by a _name prefix_, i.e. a sequence of between 0 and 256 bits, similar to how IP subnets correspond to IP address prefixes. The group `(p)` consists of all nodes whose name begins with the prefix `p`, and is responsible for all data items whose name begins with `p`. The group thus manages the part of the name space given by the interval `[p00...00, p11...11]`.
+
+When the network is bootstrapped, there is only one group, with the empty prefix `()`, responsible for the whole name space.
+
+* Whenever `(p0)` and `(p1)` satisfy certain group requirements (see below), a group `(p)` splits into `(p0)` and `(p1)`.
+* Whenever a group `(p0)` or `(p1)` ceases to satisfy the group requirements, it merges with all its sister groups into the group `(p)` again: All groups that would be a subset of `(p)` are merged back into `(p)`.
+
+These are the only two operations allowed on groups, so it is guaranteed that every address in the network belongs to exactly one group, and that group satisfies the requirements.
+
+The invariant that needs to be satisfied by the routing table is modified accordingly:
+
+1. A node must have its complete group `(p)` in its routing table.
+2. It must be connected to every group `(q)`, for which `p` and `q` differ in exactly one bit.
+
+If `p` and `q` are not of equal length, that does not count as a difference: A differing bit is one that is defined in _both_ prefixes, but is 1 in one of them and 0 in the other. So e. g. `0000` and `001000` differ in exactly one bit.
+
+The groups that differ in the `i`-th bit are the "`i`-th bucket" of `(p)`.
+
+In a balanced network, point 2 just means that the group `(q)` _is_ the `i`-th bucket. But if it is not perfectly balanced, it might be the two groups `(q0)` and `(q1)` or even more groups with longer prefixes, or some prefix of `q` itself.
+
+The requirement is symmetric: I need you in my routing table if and only if you need me in yours!
```

Comment:
This is a much stronger mechanism as well I feel and perhaps is another advantage over current?